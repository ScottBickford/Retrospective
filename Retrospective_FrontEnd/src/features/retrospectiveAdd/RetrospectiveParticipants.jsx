import React from 'react';

const RetrospectiveParticipants = ({
  participants,
  setParticipant,
  addParticipant,
  removeParticipant
}) => {
  return (
    <div>
      <div>Participants:</div>
      {participants.filter(p => !p.removed).map(p => {
        return (
          <div key={p.id}>
            <input
              type='text'
              style={{ width: 200 }}
              maxLength='200'
              value={p.name}
              onChange={e => setParticipant(p.id, e.target.value)}
            />
            <a href='/' onClick={(e) => removeParticipant(e, p.id)} style={{ marginTop: 10 }}>
              Remove
            </a>
          </div>
        );
      })}
      <div>
        <a href='/' onClick={addParticipant} style={{ marginTop: 10 }}>
          Add Participant
        </a>
      </div>
    </div>
  );
};

export default RetrospectiveParticipants;
