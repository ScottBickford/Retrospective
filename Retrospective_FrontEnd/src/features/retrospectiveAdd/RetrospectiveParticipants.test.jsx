import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import RetrospectiveParticipants from "./RetrospectiveParticipants";

const participants = [{id: 1, name: 'Person 1', removed: false}];
describe('RetrospectiveParticipants', () => {
  test('snapshot renders', () => {    
    const component = renderer.create(<RetrospectiveParticipants participants={participants} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('passes all props to Input', () => {
    const { getByText } = render(<RetrospectiveParticipants participants={participants} />);
    const buttonElement = getByText(/Participants/i);
    expect(buttonElement).toBeInTheDocument();
  });
});