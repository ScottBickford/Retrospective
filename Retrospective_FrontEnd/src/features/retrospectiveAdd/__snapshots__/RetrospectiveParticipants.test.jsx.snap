// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RetrospectiveParticipants snapshot renders 1`] = `
<div>
  <div>
    Participants:
  </div>
  <div>
    <input
      maxLength="200"
      onChange={[Function]}
      style={
        Object {
          "width": 200,
        }
      }
      type="text"
      value="Person 1"
    />
    <a
      href="/"
      onClick={[Function]}
      style={
        Object {
          "marginTop": 10,
        }
      }
    >
      Remove
    </a>
  </div>
  <div>
    <a
      href="/"
      style={
        Object {
          "marginTop": 10,
        }
      }
    >
      Add Participant
    </a>
  </div>
</div>
`;
